docs/concurrency-adoption-guidelines.md

```diff
@@ -0,0 +1,179 @@
+
+```swift
+struct Container<Value: Sendable>: Sendable { ... }
+```
+
+Here, the `Value` type must be marked `Sendable` for Swift 6’s concurrency checks to work properly with such container. However, since the Sendable type does not exist in Swift prior to Swift 5.5, it would be difficult to maintain a library that supports both Swift 5.4+ as well as Swift 6.
+
+In such situations, it may be helpful to utilize the following trick to be able to share the same Container declaration between both Swift versions of the library:
+
+```swift
+#if compiler(>=5.5)
+public typealias MYPREFIX_Sendable = Swift.Sendable
+#else 
+public typealias MYPREFIX_Sendable = Any
+#endif
+```
+
+The Any alias is effectively a no-op when applied as generic constraint, and thus this way it is possible to keep the same Container<Value> declaration working across Swift versions.
+
+### Task Local Values and Logging
+
+The newly introduced TaskLocal Values API ([SE-0311][SE-0311]) allows for implicit carrying of metadata along with Task execution. It is a natural fit for for tracing and carrying metadata around with task execution, and e.g. including it in log messages. 
+
+We are working on adjusting Swift Log to become powerful enough to automatically pick up and log specific task local values. This change will be introduced in a source compatible way. 
+
+For now libraries should continue using logger metadata, but we expect that a lot of the manually passing metadata to each log statement can be replaced in the future with setting task local values. 
+
+### Preparing for the concept of Deadlines
+
+Deadlines are another feature that closely relate to Swift Concurrency, and were originally pitched during the early versions of the Structured Concurrency proposal and later on moved out of it. The Swift team remains interested in introducing deadline concepts to the language and some preparation for it already has been performed inside the concurrency runtime. Right now however, there is no support for deadlines in Swift Concurrency and it is fine to continue using mechanisms like NIODeadline or similar mechanisms to cancel tasks after some period of time has passed. 
+
+Once Swift Concurrency gains deadline support, they will manifest in being able to cancel a task (and its child tasks) once such deadline (point in time) has been exceeded. For APIs to be “ready for deadlines” they don’t have to do anything special other than preparing to be able to deal with Tasks and their cancellation.
+
+### Cooperatively handling Task cancellation
+
+Task cancellation exists today in Swift Concurrency and is something that libraries may already handle. In practice it means that any asynchronous function (or function which is expected to be called from within Tasks), may use the Task.isCancelled or try Task.checkCancellation() APIs to check if the task it is executing in was cancelled, and if so, it may cooperatively abort any operation it was currently performing.
```

Comment:
```suggestion
`Task` cancellation exists today in Swift Concurrency and is something that libraries may already handle. In practice it means that any asynchronous function (or function which is expected to be called from within `Task`s), may use the [`Task.isCancelled`](https://developer.apple.com/documentation/swift/task/3814832-iscancelled) or [`try Task.checkCancellation()`](https://developer.apple.com/documentation/swift/task/3814826-checkcancellation) APIs to check if the task it is executing in was cancelled, and if so, it may cooperatively abort any operation it was currently performing.
```